src/main/kotlin/uk/gov/dluhc/printapi/service/PrintRequestsFileProducer.kt

```diff
@@ -111,19 +111,30 @@ class PrintRequestsFileProducer {
                 eroDeliveryTownEn ?: "",
                 eroDeliveryAreaEn ?: "",
                 eroDeliveryPostcodeEn,
-                eroNameCy ?: "",
-                eroPhoneNumberCy ?: "",
-                eroEmailAddressCy ?: "",
-                eroWebsiteCy ?: "",
-                eroDeliveryStreetCy ?: "",
-                eroDeliveryPropertyCy ?: "",
-                eroDeliveryLocalityCy ?: "",
-                eroDeliveryTownCy ?: "",
-                eroDeliveryAreaCy ?: "",
-                eroDeliveryPostcodeCy ?: "",
+                getWelshValue(eroNameCy, eroNameEn),
+                getWelshValue(eroPhoneNumberCy, eroPhoneNumberEn),
+                getWelshValue(eroEmailAddressCy, eroEmailAddressEn),
+                getWelshValue(eroWebsiteCy, eroWebsiteEn),
+                getWelshValue(eroDeliveryStreetCy, eroDeliveryStreetEn),
+                getWelshValue(eroDeliveryPropertyCy, eroDeliveryPropertyEn),
+                getWelshValue(eroDeliveryLocalityCy, eroDeliveryLocalityEn),
+                getWelshValue(eroDeliveryTownCy, eroDeliveryTownEn),
+                getWelshValue(eroDeliveryAreaCy, eroDeliveryAreaEn),
+                getWelshValue(eroDeliveryPostcodeCy, eroDeliveryPostcodeEn)
```

Comment:
I am confused with this implementation.  
e.g If an ero doesn't have a street in welsh, but has a street in english ero, then the returned address will have something like this

`Welsh Ero,
Street of English ero,
Welsh postcode
`

May be Line 114 to 123 should be done in if else loop.  i.e. if ero welsh doesn't exist, then copy over all details from english ero.

Reply:
Thanks, @vishalgupta-valtech , for the comment. 
I can understand the confusion.
Let me try to explain.
The way `getWelshValue` is implemented is, first it decides if we want to replace the existing Welsh value with the English one by checking if the certificate language is Welsh and somehow Welsh details don't exist. I decided to check the eroNameCy field to determine if the Welsh details exist or not, because I assume this value should always be there if the ERO has Welsh details (unlike the `eroDeliveryAreaCy` etc.). Then in that case we return the corresponding English value. Otherwise, it will be either the existing Welsh value or empty string if the value is null.

<pre>
    private fun PrintRequest.getWelshValue(welshValue: String?, englishValue: String?): String {
        val isWelshAndWelshPropertiesMissing =
            certificateLanguage == PrintRequest.CertificateLanguage.CY && <b>eroNameCy == null</b>

        if (isWelshAndWelshPropertiesMissing) {
            return englishValue ?: ""
        }

        return welshValue ?: ""
    }
</pre>

We might move the logic to `PrintRequestMapper` class. I am waiting @Neil-Massey's input on this.